### PR TITLE
Add PR template

### PR DESCRIPTION
**Pull request by rhett-raleigh**
## Description
This PR adds a PR template that will be used in all `encodium` repos that don't have one set. It builds on the approved template [here](https://github.com/encodium/common/pull/8857)

The goal of this PR is to increase the quality of Pull Request's created in all repos in order to simplify and enhance the PR review process.

### CODE REVIEW

**Review by drpebcak — CHANGES_REQUESTED** [review]
let's put this inside a `.github` folder to avoid cruft.

**Review by drpebcak — APPROVED** [review]

**Comment by Jesse0Michael** [comment]
Outside of the message in #dev-taco and the PR for **common** posted in #developers this has not been communicated to the developers at outside of the dev leads.

What is the problem that this solves? Is a global change the solution to it?

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+<!--- Why is this change required? What problem does it solve? -->
+
+## Testing information
+<!--- Please describe in detail how you tested your changes. -->
+<!--- This section should be through enough that reviewers can replicate the testing. -->
+
+## Issue link
+<!-- Add a link to the issue -->
+
+## Checklist before requesting a review
```
Comment (Jesse0Michael) [inline .github/pull_request_template.md:12]: These aren't applicable to every type of repository.

What are the added tests to a Terraform PR
What's the necessary observability to a GO package
What's the documentation for a Topo change


If this is a mental checklist before submitting a PR... can we write up our PR expectations.. document them in developer standards... and copy them into this template as a comment so the developer sees them when drafting the PR, but does not muddy up the PR with non-applicable text

**Comment by drpebcak** [comment]
> Outside of a message in #dev-taco and the PR for **common** posted in #developers this has not been communicated to the developers at outside of the dev leads.
> 
> What is the problem that this solves? Is a global change the solution to it?
> 
> @encodium/dev-leads

So far we have had a lot of issues with things like tests, observability, docs, etc being forgotten about while code is being written. Adding the checklist is intended to help people make sure they have considered those things before submitting the PR for review, and also to help remind reviewers to check for them.

File: .github/pull_request_template.md (hunk 1 above)
Comment (drpebcak) [reply .github/pull_request_template.md:12]: I don't think any of these can be applicable 100% of the time, no matter what the repository. If you're updating the readme, would you add tests?

It's pretty common to see these sorts of PR templates across github though, and if it's not applicable in whatever specific case, I think it's fair to just say that in the PR comment.

We already do have documentation for these standards, but people have not really adopted them. The documentation should be moved to the org readme though for sure. At this time, it is in confluence - https://revolutionparts.atlassian.net/wiki/spaces/~798050687/pages/2540405129/RevolutionParts+Engineering+Principles (yay sprawl). We did share this out though when it was written, and we've also talked about testing, observability, etc many times.

I think it's useful as a checklist for the person writing code and the reviewer as well.. These should be looked at as part of the review.

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+## Description
+<!--- Describe your changes in detail -->
+<!--- Why is this change required? What problem does it solve? -->
+
+## Testing information
+<!--- Please describe in detail how you tested your changes. -->
+<!--- This section should be through enough that reviewers can replicate the testing. -->
+
+## Issue link
+<!-- Add a link to the issue -->
```
[the remark this answers is not in the document]
Comment (drpebcak) [reply .github/pull_request_template.md:10]: Yes using the integration is probably better, and if we do it consistently, linking it in the description will not be necessary.

File: .github/pull_request_template.md (hunk 1 above)
Comment (Jesse0Michael) [reply .github/pull_request_template.md:12]: PR to move engineering principles to github: https://github.com/encodium/.github-private/pull/4

**Pull request merged**